lib/checkin_handler.py

```diff
@@ -146,9 +147,14 @@ def _check_in(self) -> None:
         except RequestError as err:
             logger.debug("Failed to check in. Error: %s. Exiting", err)
             self.notification_handler.failed_checkin(err, self.flight)
+            if self.checkin_scheduler.reservation_monitor.config.healthchecks_url is not None:
```

Comment:
I agree with your comment that these are not necessary as a notification is already sent. The Healthchecks feature can be just for ensuring the script is successfully running on the specified (or default) interval.